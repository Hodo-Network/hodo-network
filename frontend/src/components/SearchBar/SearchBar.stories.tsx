import { Meta, Story } from '@storybook/react';

import { SearchBar } from './SearchBar';

export default {
  title: 'Components/SearchBar',
  component: SearchBar,
} as Meta;

const Template: Story = (args) => <SearchBar {...args} />;

export const Default = Template.bind({});
Default.args = {};
