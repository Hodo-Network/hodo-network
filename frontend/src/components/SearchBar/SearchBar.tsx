import React from "react";
import { SearchIcon } from "@heroicons/react/solid";

export const SearchBar = () => {
  return (
    <div className='relative text-gray-400 focus-within:text-gray-500'>
      <label htmlFor='search' className='sr-only'>
        Search
      </label>
      <input
        id='search'
        type='search'
        placeholder='Search'
        className='block w-full border-transparent pl-8 placeholder-gray-500 focus:border-transparent sm:text-sm focus:ring-0 dark:bg-gray-900'
      />
      <div className='pointer-events-none absolute inset-y-0 left-0 flex items-center justify-center'>
        <SearchIcon className='h-5 w-5' aria-hidden='true' />
      </div>
    </div>
  );
};
