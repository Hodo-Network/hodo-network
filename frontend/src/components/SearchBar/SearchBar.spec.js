import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchBar from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SearchBar />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
