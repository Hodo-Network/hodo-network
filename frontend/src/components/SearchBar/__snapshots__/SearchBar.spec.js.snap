// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="max-w-2xl relative text-gray-400 focus-within:text-gray-500"
>
  <label
    className="sr-only"
    htmlFor="search"
  >
    Search
  </label>
  <input
    className="block w-full border-transparent pl-8 placeholder-gray-500 focus:border-transparent sm:text-sm focus:ring-0 dark:bg-gray-900"
    id="search"
    placeholder="Search"
    type="search"
  />
  <div
    className="pointer-events-none absolute inset-y-0 left-0 flex items-center justify-center"
  >
    <svg
      aria-hidden="true"
      className="h-5 w-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</div>
`;
